chapters/classes.tex

```diff
@@ -847,7 +847,7 @@ \subsection{Combining Base Classes and Other Elements}\label{restriction-on-comb
 
 It is not legal to combine equations, algorithms, components, non-empty base classes (see below), or protected elements with an extends from an array class, a class with non-empty \lstinline[language=grammar]!base-prefix!, a \firstuse{simple type} (\lstinline!Real!, \lstinline!Boolean!, \lstinline!Integer!, \lstinline!String! and enumeration types), or any class transitively extending from an array class, a class with non-empty \lstinline[language=grammar]!base-prefix!, or a simple type.
 
-\begin{definition}[Empty class]\index{empty-class}
+\begin{definition}[Empty class]\index{class!empty}\label{empty-class}
```

Comment:
I'd find the direct index entry more useful, doubting that we need one under _class_:
```suggestion
\begin{definition}[Empty class]\index{empty class}\label{empty-class}
```